core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java

```diff
@@ -1938,6 +1939,12 @@ public static TableInfo convertGsonToTableInfo(Map<String, String> properties) {
       }
       builder.append(part);
     }
+
+    // Datatype GSON adapter is added to support backward compatibility for tableInfo desrialization
+    GsonBuilder gsonBuilder = new GsonBuilder();
+    gsonBuilder.registerTypeAdapter(DataType.class, DataTypes.DataTypeAdapter.getInstance());
```

Comment:
I am not sure whether it has concurrent issue if we use singleton here, I think it is acceptable to new a DataTypeAdapter

Reply:
ok